app/js/components/manageApps/ManageApps.jsx

```diff
@@ -243,7 +249,26 @@ export default class ManageApps extends React.Component {
 
   handleDownload(e) {
     e.preventDefault();
-    location.href=this.state.downloadUri;
+    const xhr = new XMLHttpRequest();
+    xhr.open('GET', this.state.downloadUri, true);
+    xhr.responseType = 'blob';
+    let filename = this.state.downloadUri.split('=').slice(-1)[0];
+    let that = this;
+    
+    xhr.onload = function(e) {
+      if (this.status == 200) {
+        let data = this.response;
+        const blob = new Blob([data], {type: 'application/octet-stream'});
+        const file = new File([blob], filename, {type: 'application/zip'});
+        that.handleUpload(file);
+        this.setState((prevState, props) => {
```

Comment:
Hi noticed this.setState() throws an error because of the context of the `this` keyword. I would recommend using `that.setState` here.